File: mobile/lib/models/session.dart
/// Session 模型与 SessionStatus 枚举。
library;

import 'unified_event.dart';

enum SessionStatus {
  running('running'),
  idle('idle'),
  ended('ended'),
  error('error');

  final String wire;
  const SessionStatus(this.wire);

  static SessionStatus fromWire(String? v) =>
      values.firstWhere((e) => e.wire == v, orElse: () => SessionStatus.idle);
}

class Session {
  final String id;
  final String taskId;
  final String? cliSessionId;
  final CliKind cli;
  final String? model;
  final SessionStatus status;
  final DateTime createdAt;
  final DateTime? endedAt;

  const Session({
    required this.id,
    required this.taskId,
    this.cliSessionId,
    required this.cli,
    this.model,
    required this.status,
    required this.createdAt,
    this.endedAt,
  });

  factory Session.fromJson(Map<String, dynamic> j) => Session(
        id: j['id'] as String? ?? '',
        taskId: j['taskId'] as String? ?? '',
        cliSessionId: j['cliSessionId'] as String?,
        cli: CliKind.fromWire(j['cli'] as String?),
        model: j['model'] as String?,
        status: SessionStatus.fromWire(j['status'] as String?),
        createdAt: DateTime.tryParse(j['createdAt'] as String? ?? '') ?? DateTime.now(),
        endedAt: j['endedAt'] != null ? DateTime.tryParse(j['endedAt'] as String) : null,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'taskId': taskId,
        if (cliSessionId != null) 'cliSessionId': cliSessionId,
        'cli': cli.wire,
        if (model != null) 'model': model,
        'status': status.wire,
        'createdAt': createdAt.toIso8601String(),
        if (endedAt != null) 'endedAt': endedAt!.toIso8601String(),
      };
}

File: backend-go/internal/types/session.go
package types

import "time"

// SessionStatus 标识会话状态（见 §2.5.1）。
//
// 注意：本类型与 store.SessionStatus 是同一概念；
// 这里用 wire 层类型避免 store 包导出 SessionStatus 后 types 依赖 store。
// 实际 wire 值由 store 层 Normalize 时映射为本类型字符串。
type SessionStatus string

const (
	SessionStatusRunning SessionStatus = "running"
	SessionStatusIdle    SessionStatus = "idle"
	SessionStatusEnded   SessionStatus = "ended"
	SessionStatusError   SessionStatus = "error"
)

// NormalizeSessionStatus 把任意字符串归一为合法 SessionStatus，非法值回退 idle。
func NormalizeSessionStatus(s string) SessionStatus {
	switch SessionStatus(s) {
	case SessionStatusRunning, SessionStatusIdle, SessionStatusEnded, SessionStatusError:
		return SessionStatus(s)
	}
	return SessionStatusIdle
}

// SessionDTO 是 Session 的对外响应 DTO（见 §2.3.6）。
//
// TaskID 可空（WS 直接启动会话时为 null）；
// CliSessionID 可空（收到 session.init 后才写入）；
// EndedAt 可空（status=running/idle 时为 null）；
// Model 可空。
type SessionDTO struct {
	ID           string     `json:"id"`
	TaskID       *string    `json:"taskId,omitempty"`
	CliSessionID string     `json:"cliSessionId,omitempty"`
	Cli          CliKind    `json:"cli"`
	Model        string     `json:"model,omitempty"`
	Status       string     `json:"status"`
	CreatedAt    time.Time  `json:"createdAt"`
	EndedAt      *time.Time `json:"endedAt,omitempty"`
}

// CreateSessionParams 是 POST /api/sessions 的入参（§2.3.6）。
// TaskID 可空（WS 直接启动会话时为 null）；Model 可空。
type CreateSessionParams struct {
	TaskID *string `json:"taskId,omitempty"`
	Cli    CliKind `json:"cli"`
	Model  string  `json:"model,omitempty"`
}
